sdk/metric/periodic_reader.go

```diff
@@ -206,9 +207,10 @@ func (r *periodicReader) aggregation(kind InstrumentKind) aggregation.Aggregatio
 // collectAndExport gather all metric data related to the periodicReader r from
 // the SDK and exports it with r's exporter.
 func (r *periodicReader) collectAndExport(ctx context.Context) error {
-	m, err := r.Collect(ctx)
+	rm := metricdata.ResourceMetrics{}
```

Comment:
```suggestion
	// TODO (#3047): Use a sync.Pool or persistent pointer instead of allocating rm every Collect.
	rm := metricdata.ResourceMetrics{}
```